### PR TITLE
[student-libraries] Create S3 API for publishing/using student libraries

### PR DESCRIPTION
**Pull request by jmkulwik**
# Description
Provides PUT, GET, and DELETE access to the cdo-v3-libraries bucket.

### Usage
let api = new LibraryClientApi(channelId);
api.getLatest();
### Future Work
Create import and publish capabilities and UX for applab projects

## Links

- [spec](https://docs.google.com/document/d/1VxMmdHdIeBWCXKofkjEy9MuWcrzeEW3Zt-Ir8pINYdA/edit#heading=h.9a8gdpezdgnm)
- [jira](https://codedotorg.atlassian.net/browse/STAR-662)

## Testing story
Added tests for s3 api. Mostly did pattern matching on tests for animations and sources apis.

# Reviewer Checklist:

- [x] Tests provide adequate coverage
- [x] Code is well-commented
- [x] New features are translatable or updates will not break translations
- [x] Relevant documentation has been added or updated
- [x] User impact is well-understood and desirable
- [x] Pull Request is labeled appropriately
- [x] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by jmkulwik** [comment]
Note: The fixtures files are all autogenerated files that are supposed to get checked in to the repo. For more details see here: https://github.com/code-dot-org/code-dot-org/blob/staging/shared/test/README.md

File: shared/middleware/files_api.rb [hunk 1]
```diff
@@ -395,10 +397,11 @@ def copy_file(endpoint, encrypted_channel_id, filename, source_filename)
   #
   # PUT /v3/(sources)/<channel-id>/<filename>?version=<version-id>
   # PUT /v3/(assets)/<channel-id>/<filename>
+  # PUT /v3/(libraries)/<channel-id>/<filename>
```
Comment (epeach) [inline shared/middleware/files_api.rb]: Super-nit - can we update this comment style to match how we do the others in this file?

File: apps/src/code-studio/components/Libraries/LibraryClientApi.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    );
+  }
+
+  getVersion(versionId) {
+    let library;
+    this.libraryApi.fetch(
+      this.channelId + '/' + LIBRARY_NAME + '?version=' + versionId,
+      (error, data) => {
+        if (data) {
+          // in the future, responses will be passed back to the import dialog
```
Comment (epeach) [inline apps/src/code-studio/components/Libraries/LibraryClientApi.js:45]: Are these ToDos tracked?

File: shared/test/test_libraries.rb [hunk 3]
```diff
@@ -0,0 +1,143 @@
+require_relative 'files_api_test_base' # Must be required first to establish load paths
+require_relative 'files_api_test_helper'
+
+class LibrariesTest < FilesApiTestBase
```
Comment (epeach) [inline shared/test/test_libraries.rb:4]: :tada: Nice tests! :tada:

File: apps/src/code-studio/components/Libraries/LibraryClientApi.js (hunk 2 above)
Comment (jmkulwik) [reply apps/src/code-studio/components/Libraries/LibraryClientApi.js:45]: Yes - this is just setting up some basic scaffolding. I expect this to evolve a bit as the libraries feature goes through a couple churns. I'll expect to finish these up as part of these two tasks:
https://codedotorg.atlassian.net/browse/STAR-710
https://codedotorg.atlassian.net/browse/STAR-709

**Review by epeach — APPROVED** [review]

File: shared/middleware/files_api.rb (hunk 1 above)
Comment (jmkulwik) [reply shared/middleware/files_api.rb]: Fixed!

**Pull request merged**